src/Hydra/Serializer/ConstraintViolationListNormalizer.php

```diff
@@ -45,11 +45,16 @@ public function normalize($object, $format = null, array $context = [])
         $messages = [];
 
         foreach ($object as $violation) {
-            $violations[] = [
+            $violationData = [
                 'propertyPath' => $violation->getPropertyPath(),
-                'message' => $violation->getMessage(),
+                'message' => $violation->getMessage()
             ];
 
+            if (! empty($violation->getConstraint()->payload)) {
```

Comment:
I suggest to store `$violation->getConstraint()->payload` in a variable to prevent an extra function call. Btw you can remove the call to `empty`, it is useless.

Reply:
Done ! Thanks